api/src/features/todo/use_cases/update_todo.py

```diff
@@ -23,10 +24,14 @@ class UpdateTodoResponse(BaseModel):
 def update_todo_use_case(
     id: str,
     data: UpdateTodoRequest,
+    user_id: str,
     todo_repository: TodoRepositoryInterface,
 ) -> UpdateTodoResponse:
     todo_item = todo_repository.get(id)
-    updated_todo_item = TodoItem(id=todo_item.id, title=data.title, is_completed=data.is_completed)
+    if todo_item.user_id != user_id:
+        raise NotFoundException
```

Comment:
Is NotFoundException correct? or use MissingPrivilegeException, e.g 403 Forbidden – client authenticated but does not have permission to access the requested resource.

Reply:
(Btw, this only makes sense if we don't have sequential ids, because if we do - he can just create one todo himself, get the id (e.g. 102398) and know that every todo id below that exists or has existed.)

Reply:
@soofstad @netr0m any comment on security standpoint, see above

Reply:
I see where you're coming from @eoaksnes , and in general I'd agree - however in this case it would lead to (non-trivial) information disclosure and as such returning a `404 Not Found` makes more sense IMO.

Reply:
I say we should try and have an API that respond as expected, "proper status codes".
The only reason to hide a 403 behind a 404 is if you give away to much information with a 403.

If the only information given is that "a TODO with this uid exists", there should be no risk with that

Reply:
*A short huddle later.* 

I'll implement this using 403's just to make it clear that this Exception/status-code can be used. We're not divulging any significant information, so it's not a problem.